libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DHCP.c

```diff
@@ -321,45 +309,54 @@ BaseType_t xGivingUp = pdFALSE;
 			{
 			#if( ipconfigUSE_DHCP_HOOK != 0 )
 				/* Ask the user if a DHCP request is required. */
-				eAnswer = xApplicationDHCPHook( eDHCPPhasePreRequest, xDHCPData.ulOfferedIPAddress );
+				eAnswer = xApplicationDHCPHook( eDHCPPhasePreRequest, EP_DHCPData.ulOfferedIPAddress );
 
 				if( eAnswer == eDHCPContinue )
 			#endif	/* ipconfigUSE_DHCP_HOOK */
 				{
 					/* An offer has been made, the user wants to continue,
 					generate the request. */
-					xDHCPData.xDHCPTxTime = xTaskGetTickCount();
-					xDHCPData.xDHCPTxPeriod = dhcpINITIAL_DHCP_TX_PERIOD;
-					prvSendDHCPRequest( );
-					xDHCPData.eDHCPState = eWaitingAcknowledge;
+					EP_DHCPData.xDHCPTxTime = xTaskGetTickCount();
+					EP_DHCPData.xDHCPTxPeriod = dhcpINITIAL_DHCP_TX_PERIOD;
+					prvSendDHCPRequest();
+					EP_DHCPData.eDHCPState = eWaitingAcknowledge;
 					break;
 				}
 
 			#if( ipconfigUSE_DHCP_HOOK != 0 )
 				if( eAnswer == eDHCPUseDefaults )
 				{
-					memcpy( &xNetworkAddressing, &xDefaultAddressing, sizeof( xNetworkAddressing ) );
+					( void ) memcpy( &( xNetworkAddressing ), &( xDefaultAddressing ), sizeof( xNetworkAddressing ) );
 				}
 
 				/* The user indicates that the DHCP process does not continue. */
 				xGivingUp = pdTRUE;
 			#endif	/* ipconfigUSE_DHCP_HOOK */
 			}
-			else if( ( xTaskGetTickCount() - xDHCPData.xDHCPTxTime ) > xDHCPData.xDHCPTxPeriod )
+
+			/* Is it time to send another Discover? */
+			else if( ( xTaskGetTickCount() - EP_DHCPData.xDHCPTxTime ) > EP_DHCPData.xDHCPTxPeriod )
 			{
 				/* It is time to send another Discover.  Increase the time
 				period, and if it has not got to the point of giving up - send
 				another discovery. */
-				xDHCPData.xDHCPTxPeriod <<= 1;
+				EP_DHCPData.xDHCPTxPeriod <<= 1;
 
-				if( xDHCPData.xDHCPTxPeriod <= ipconfigMAXIMUM_DISCOVER_TX_PERIOD )
+				if( EP_DHCPData.xDHCPTxPeriod <= ipconfigMAXIMUM_DISCOVER_TX_PERIOD )
 				{
-					if( xApplicationGetRandomNumber( &( xDHCPData.ulTransactionId ) ) != pdFALSE )
+					if( xApplicationGetRandomNumber( &( EP_DHCPData.ulTransactionId ) ) != pdFALSE )
 					{
-						xDHCPData.xDHCPTxTime = xTaskGetTickCount( );
-						xDHCPData.xUseBroadcast = !xDHCPData.xUseBroadcast;
-						prvSendDHCPDiscover( );
-						FreeRTOS_debug_printf( ( "vDHCPProcess: timeout %lu ticks\n", xDHCPData.xDHCPTxPeriod ) );
+						EP_DHCPData.xDHCPTxTime = xTaskGetTickCount();
+						if( EP_DHCPData.xUseBroadcast != pdFALSE )
+						{
+							EP_DHCPData.xUseBroadcast = pdFALSE;
+						}
+						else
+						{
+							EP_DHCPData.xUseBroadcast = pdTRUE;
+						}
```

Comment:
What is the rationale behind not using `EP_DHCPData.xUseBroadcast = ! EP_DHCPData.xUseBroadcast` as done in earlier code? Why should we introduce branches?

Reply:
PC-lint doesn't like expressions with NOT as in `!( EP_DHCPData.xUseBroadcast )`.

Reply:
Ah I see... I just think that the readablility is not as good as the previous case. But of course, this is not a blocking issue.